torch/distributed/_tensor/dispatch.py

```diff
@@ -82,24 +82,29 @@ def redistribute_local_args(
 
     # TODO: the op schema should probably just remain flattened so that we can avoid this tree flatten
     # Need to fix all the ops before doing this.
-    flatten_args_schema_to_reshard = tree_flatten(suggested_input_schema.args_schema)[0]
+    if op_info.args_tree_spec is not None:
+        flatten_args_schema_to_reshard = tree_flatten(
+            suggested_input_schema.args_schema
+        )[0]
+    else:
+        flatten_args_schema_to_reshard = suggested_input_schema.args_schema
 
-    new_flat_local_args: List[object] = []
+    new_local_args: List[object] = []
     for i, arg_spec in enumerate(op_info.flat_args_schema):
         reshard_arg_spec = flatten_args_schema_to_reshard[i]
         if isinstance(arg_spec, DTensorSpec):
-            local_tensor = cast(torch.Tensor, op_info.flat_local_args[i])
+            local_tensor = cast(torch.Tensor, op_info.local_args[i])
             if arg_spec != reshard_arg_spec:
                 resharded_local_tensor = redistribute_local_tensor(
                     local_tensor, arg_spec, reshard_arg_spec
                 )
-                new_flat_local_args.append(resharded_local_tensor)
+                new_local_args.append(resharded_local_tensor)
             else:
-                new_flat_local_args.append(local_tensor)
+                new_local_args.append(local_tensor)
         else:
-            new_flat_local_args.append(reshard_arg_spec)
+            new_local_args.append(reshard_arg_spec)
 
-    op_info.flat_local_args = new_flat_local_args
```

Comment:
it's interesting it worked without `tuple()` previously..

Reply:
Yeah it's because we always to unflatten before, which recovers the list to tuple